### DISCUSSION TITLE
MSBasic Or access to open and close files from basic

### DISCUSSION THREAD

**Discussion by Bee91303** [picocomputer/community#109] [Q&A]
I would like to make a contacts program in Basic.  For this to be feasible you need a way to store and retrieve data from inside the program.  I understand the objection to MSbasic but is anyone work on either of these?  Porting MSbasic or adding the functionality tr EHbasic?

Thx

**Comment by netzerohero** [comment]
Have you considered using EHBasic's "DATA" keyword (to store your contact data directly within BASIC source code) along with using "READ" and "RESTORE"?

However, if you are considering file read/write access from EHbasic: I don't believe a file read/write feature from within EHbasic is currently available.  Neither the [EHbasic](https://github.com/picocomputer/ehbasic) nor [EHbasic-plus](https://github.com/picocomputer/ehbasic-plus) have that level of maturity.  EHBasic-plus would be the place to first try adding those "experimental" features.

You may want to refer to this [discussion](https://github.com/orgs/picocomputer/discussions/82); see the two open rp6502 API file-access features [11](https://github.com/picocomputer/rp6502/issues/11) and [12](https://github.com/picocomputer/rp6502/issues/12).  Adding OS (API) calls from within either EHbasic repo would then theoretically provide file-level access from within EHbasic.
